src/plugins/controls/public/control_types/range_slider/range_slider_embeddable.tsx

```diff
@@ -0,0 +1,312 @@
+import { ControlsDataService } from '../../services/data';
+import { ControlInput, ControlOutput } from '../..';
+import { pluginServices } from '../../services';
+
+import { RangeSliderComponent, RangeSliderComponentState } from './range_slider.component';
+import { rangeSliderReducers } from './range_slider_reducers';
+import { RangeSliderStrings } from './range_slider_strings';
+import { RangeSliderEmbeddableInput, RANGE_SLIDER_CONTROL } from './types';
+
+const RangeSliderReduxWrapper = withSuspense<
+  ReduxEmbeddableWrapperPropsWithChildren<RangeSliderEmbeddableInput>
+>(LazyReduxEmbeddableWrapper);
+
+const diffDataFetchProps = (
+  current?: RangeSliderDataFetchProps,
+  last?: RangeSliderDataFetchProps
+) => {
+  if (!current || !last) return false;
+  const { filters: currentFilters, ...currentWithoutFilters } = current;
+  const { filters: lastFilters, ...lastWithoutFilters } = last;
+  if (!deepEqual(currentWithoutFilters, lastWithoutFilters)) return false;
+  if (!compareFilters(lastFilters ?? [], currentFilters ?? [], COMPARE_ALL_OPTIONS)) return false;
+  return true;
+};
+
+interface RangeSliderDataFetchProps {
+  fieldName: string;
+  dataViewId: string;
+  query?: ControlInput['query'];
+  filters?: ControlInput['filters'];
+}
+
+const fieldMissingError = (fieldName: string) =>
+  new Error(`field ${fieldName} not found in index pattern`);
+
+export class RangeSliderEmbeddable extends Embeddable<RangeSliderEmbeddableInput, ControlOutput> {
+  public readonly type = RANGE_SLIDER_CONTROL;
+  public deferEmbeddableLoad = true;
+
+  private subscriptions: Subscription = new Subscription();
+  private node?: HTMLElement;
+
+  // Controls services
+  private dataService: ControlsDataService;
+  private dataViewsService: ControlsDataViewsService;
+
+  // Internal data fetching state for this input control.
+  private dataView?: DataView;
+  private field?: DataViewField;
+
+  // State to be passed down to component
+  private componentState: RangeSliderComponentState;
+  private componentStateSubject$ = new BehaviorSubject<RangeSliderComponentState>({
+    min: '',
+    max: '',
+    loading: true,
+    fieldFormatter: (value: string) => value,
+  });
+
+  constructor(input: RangeSliderEmbeddableInput, output: ControlOutput, parent?: IContainer) {
+    super(input, output, parent); // get filters for initial output...
+
+    // Destructure controls services
+    ({ data: this.dataService, dataViews: this.dataViewsService } = pluginServices.getServices());
+
+    this.componentState = {
+      min: '',
+      max: '',
+      loading: true,
+      fieldFormatter: (value: string) => value,
+    };
+    this.updateComponentState(this.componentState);
+
+    this.initialize();
+  }
+
+  private initialize = async () => {
+    const initialValue = this.getInput().value;
+    if (initialValue) {
+      await this.fetchMinMax();
+    }
+    this.setInitializationFinished();
+    this.setupSubscriptions();
```

Comment:
I noticed a small bug here where the minMax is fetched twice on startup. Not exactly sure how to fix it, but I think it's caused here.